### ISSUE TITLE
Model evaluation yields incoherent confusion matrix results

### ISSUE THREAD

**Issue by lorenzo0507** [ultralytics/ultralytics#25724]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

When running model.val() in Ultralytics versions >= 8.4.112 (possibly earlier), the results for Confusion Matrix and Normalized Confusion Matrix are incoherent.

These are the validation results for a previously trained model on a dataset that has 160 validation images.

<img width="3000" height="2250" alt="Image" src="https://github.com/user-attachments/assets/688e53f6-5ef5-4670-adf3-dad81f57da01" />

<img width="3000" height="2250" alt="Image" src="https://github.com/user-attachments/assets/6e82023d-78ee-45b5-8079-4454849e5231" />

The results make little sense - Class Istrice has more false positives than there are images in the validation set, wrong-class predictions are much more frequent than previous versions (see below) and background predictions are ridiculously high across the board.

I have validated the same model on version 8.4.101 and output matches expectations:

<img width="3000" height="2250" alt="Image" src="https://github.com/user-attachments/assets/fac7619a-68aa-4954-9576-9aee3e70e06f" />

<img width="3000" height="2250" alt="Image" src="https://github.com/user-attachments/assets/ab53a890-813f-4ed0-952e-4cd9acffac29" />

This model had been trained on a non-affected version of Ultralytics and exported in NCNN format, but i can confirm this also happens with newly trained models and other formats (can confirm HEF is affected).

### Environment

Ultralytics 8.4.116 🚀 Python-3.10.20 torch-2.13.0+cu130 CUDA:0 (NVIDIA GeForce RTX 4070 SUPER, 12282MiB)
Setup complete ✅ (16 CPUs, 15.6 GB RAM, 112.4/1006.9 GB disk)

OS                     Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.39
Environment            Linux
Python                 3.10.20
Install                git
Path                   /mnt/c/Users/(redacted)/.venv/lib/python3.10/site-packages/ultralytics
RAM                    15.62 GB
Disk                   112.4/1006.9 GB
CPU                    AMD Ryzen 7 7700X 8-Core Processor
CPU count              16
GPU                    NVIDIA GeForce RTX 4070 SUPER, 12282MiB
GPU count              1
CUDA                   13.0

filelock               ✅ 3.32.2>=3.16.1
numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.9>=3.3.0
opencv-python          ✅ 5.0.0.93!=4.13.0.90,>=4.7.0
pillow                 ✅ 12.3.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.34.2>=2.23.0
torch                  ✅ 2.13.0>=1.8.0
torch                  ✅ 2.13.0!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.28.0>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.43.2>=0.20.0
nvidia-ml-py           ✅ 13.610.43>=12.0.0
ultralytics-thop       ✅ 2.1.6>=2.1.6

### Minimal Reproducible Example

To reproduce, confront the (Normalized) Confusion Matrix output of any model.val() in affected versions (ie. latest/8.4.116) with those from a non-affected version (ie. 8.4.101).
```python3
model = YOLO(model_path, task="detect")
results = model.val(data="./dataset/data.yaml")
print(results.confusion_matrix.matrix)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lorenzo0507, thank you for reporting the inconsistent confusion matrix and normalized confusion matrix results between Ultralytics 8.4.101 and 8.4.116. This automated response confirms that an Ultralytics engineer will assist soon.

To help isolate the regression, please provide the complete `model.val()` arguments and the raw `results.confusion_matrix.matrix` output from both versions, along with a shareable minimal dataset or sample data reproducing the issue. A [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) is especially helpful.

Please also verify the behavior with the latest `ultralytics` package and include any focused validation logs if the issue persists.

**Comment by Y-T-G** [comment]
Thanks for reporting.

This was fixed recently. It will be available in the next release.
